datafusion/sql/src/expr/identifier.rs

```diff
@@ -44,18 +49,49 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             // interpret names with '.' as if they were
             // compound identifiers, but this is not a compound
             // identifier. (e.g. it is "foo.bar" not foo.bar)
-
-            Ok(Expr::Column(Column {
-                relation: None,
-                name: normalize_ident(id),
-            }))
+            let normalize_ident = normalize_ident(id);
+            match schema.field_with_unqualified_name(normalize_ident.as_str()) {
+                Ok(_) => {
+                    // found a match without a qualified name, this is a inner table column
+                    Ok(Expr::Column(Column {
+                        relation: None,
+                        name: normalize_ident,
+                    }))
+                }
+                Err(_) => {
+                    let outer_query_schema_opt =
+                        planner_context.outer_query_schema.clone();
+                    if let Some(outer) = outer_query_schema_opt.as_ref() {
```

Comment:
Is there a reason for this clone (of the entire schema)? Could it be more like

```suggestion
                    let outer_query_schema_opt = planner_context.outer_query_schema.as_ref();
                    if let Some(outer) = outer_query_schema_opt {
```

Reply:
The same thing applies to the other places in this file